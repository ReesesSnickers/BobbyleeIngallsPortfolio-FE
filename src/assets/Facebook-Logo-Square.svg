<?xml version="1.0" encoding="utf-8"?>
<!-- www.vectorico.com -->
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 64 64" style="enable-background:new 0 0 64 64;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#3A559F;}
	.st1{fill:#FFFFFF;}
</style>
<title>facebook logo</title>
<rect class="st0" width="64" height="64"/>
<path class="st1" d="M36.2,59.2V34.4h8.3l1.2-9.7h-9.6v-6.2c0-2.8,0.8-4.7,4.8-4.7l5.1,0V5.2c-0.9-0.1-3.9-0.4-7.5-0.4
	c-7.4,0-12.4,4.5-12.4,12.8v7.1h-8.3v9.7h8.3v24.8H36.2z"/>
</svg>
