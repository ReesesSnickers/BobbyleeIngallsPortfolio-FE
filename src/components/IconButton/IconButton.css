.portfolio-IconButton {
  width: 80px;
  height: 80px;
  border-radius: 13px;
  padding: 0px;
  border: none;
  cursor: pointer;
}
.portfolio-IconButton svg {
  border-radius: 10px;
}
.portfolio-IconButton:focus {
  width: 82px;
  height: 82px;
  /* outline: none; */
  border: 2px solid #eb21d4;
  box-shadow: 0px 0px 5px #eb21d4;
}
