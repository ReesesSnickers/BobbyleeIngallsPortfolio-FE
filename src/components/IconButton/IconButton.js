import React from "react";
import PropTypes from "prop-types";

const IconButton = ({ icon, href, social }) => {
  return (
    <a
      href={href}
      target="_blank"
      rel="noopener noreferrer"
      aria-label={`Redirect to Bobbylee's ${social} profile`}
      className="portfolio-IconButton"
    >
      {icon}
    </a>
  );
};

export default IconButton;

IconButton.propTypes = {
  // icon to be shown
  icon: PropTypes.node.isRequired,
  // url string for the button to navigate to
  href: PropTypes.string,
  // what social media source is provided
  social: PropTypes.string
};
