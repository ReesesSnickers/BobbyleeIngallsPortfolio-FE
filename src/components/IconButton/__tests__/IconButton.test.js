import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import IconButton from "../IconButton.js";

import { ReactComponent as Facebook } from "../../../assets/Facebook-Logo-Square.svg";

let props;
let mountedComponent;

describe("IconButton", () => {
  const testComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<IconButton {...props} />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    props = {
      icon: <Facebook />,
      href: undefined
    };
    mountedComponent = undefined;
  });

  // initial render
  it("Renders the IconButton component", () => {
    let button = testComponent().find("a.portfolio-IconButton");
    expect(button.length).toBe(1);
  });

  // snapshot
  it("should match snapshot", () => {
    const tree = renderer.create(<IconButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // icon
  it("Renders the Icon in the ancors inner text", () => {
    let button = testComponent().find("a");
    expect(button.text()).toBe("Facebook-Logo-Square.svg");
  });

  // href
  describe("When href is defined", () => {
    beforeEach(() => {
      props.href = "MockHref";
    });
    it("Should pass the href prop to the ancors href prop", () => {
      const component = testComponent().find("a");
      expect(component.props().href).toBe("MockHref");
    });
  });
});
