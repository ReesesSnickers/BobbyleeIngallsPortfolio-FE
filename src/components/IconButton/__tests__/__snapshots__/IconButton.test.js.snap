// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match snapshot 1`] = `
<a
  className="portfolio-IconButton"
  rel="noopener noreferrer"
  target="_blank"
>
  <svg>
    Facebook-Logo-Square.svg
  </svg>
</a>
`;
